src/test/java/com/artipie/pypi/PypiITCase.java

```diff
@@ -23,205 +23,95 @@
+                "/var/artipie/data/artipie/pypi/example-pckg",
+                BindMode.READ_ONLY
             )
+    );
+
+    @BeforeEach
+    void setUp() throws IOException {
+        this.containers.assertExec(
+            "Apt-get update failed",
+            new MavenITCase.ContainerResultMatcher(),
+            "apt-get", "update"
         );
-        MatcherAssert.assertThat(
-            "Packages should be saved in storage",
-            this.inStorage("artipietestpkg-0.0.3-py2-none-any.whl")
-                && this.inStorage("artipietestpkg-0.0.3.tar.gz"),
-            new IsEqual<>(true)
+        this.containers.assertExec(
+            "Failed to install twine",
+            new MavenITCase.ContainerResultMatcher(),
+            "python", "-m", "pip", "install", "twine"
         );
-    }
-
-    @Test
-    void pypiInstallShouldFailWithForbidden() throws Exception {
-        this.init(false);
-        MatcherAssert.assertThat(
-            this.cntn.execStdoutWithoutCheckExitCode(
-                "pip", "install", "--verbose", "--no-deps", "--trusted-host", PypiITCase.HOST,
-                "--index-url", this.url.string(ArtipieServer.BOB), "anypackage"
-            ),
-            new StringContains(
-                String.format(
-                    "403 Client Error: Forbidden for url: %spip/", this.url.string()
-                )
-            )
+        this.containers.assertExec(
+            "Failed to upgrade pip",
+            new MavenITCase.ContainerResultMatcher(),
+            "python", "-m", "pip", "install", "--upgrade", "pip"
         );
     }
 
     @Test
-    void pypiPublishShouldFailSinceForbidden() throws Exception {
-        this.init(false);
-        new TestResource("pypi-repo/example-pckg/dist/")
-            .addFilesTo(
-                this.storage,
-                new Key.From("pypi-repo", "example-pckg")
+    void installPythonPackage() throws IOException {
+        String meta = "pypi-repo/example-pckg/dist/artipietestpkg-0.0.3.tar.gz";
+        this.containers.putResourceToArtipie(
+            meta, "/var/artipie/data/my-python/artipietestpkg/artipietestpkg-0.0.3.tar.gz"
         );
-        this.cntn.execStdout("python3", "-m", "pip", "install", "--user", "--upgrade", "twine");
-        MatcherAssert.assertThat(
-            "Packages should not be uploaded",
-            this.cntn.execStdErr(
-                "python3", "-m", "twine", "upload", "--verbose",
-                "--repository-url", this.url.string(),
-                "-u", ArtipieServer.BOB.name(), "-p", ArtipieServer.BOB.password(),
-                "pypi-repo/example-pckg/*"
-            ),
-            new StringContains("HTTPError: 403 Forbidden")
+        meta = "pypi-repo/example-pckg/dist/artipietestpkg-0.0.3-py2-none-any.whl";
+        this.containers.putResourceToArtipie(
+            meta,
+            "/var/artipie/data/my-python/artipietestpkg/artipietestpkg-0.0.3-py2-none-any.whl"
```

Comment:
@milkysoft there is no need to put this `whl` to artipie as it is intended for python 2 (container with python:3.7 is used in this test, python 2 and 3 are not compatible), `tar.gz` should be enoght. Also, seems like variable `meta` can be inlined.

Reply:
@olenagerasimova whl is removed with latest commit, `meta` declared `final`